### DISCUSSION TITLE
Surface editing

### DISCUSSION THREAD

**Discussion by portas525** [pupil-labs/pupil#2280] [Ideas]
Hello,
I´m new at github and can´t contirubte here with writing code, but I habe a lot of experience in eyetracking with other eyetracker (ergoneers, tobi, smi, smarteye). 
My question:
Is there an "easy" possibility to add an button or something to edit a drawn surface easier? 
When you open the offline surface tracker (e.g. in the playser) there it will be nice when there appears a button to choose the surface (and highlight it at the screen).
I hope that this post is not a double post.
Thank you.
Best
portas525

**Comment by willpatera** [comment]
Hi @portas525 :wave: Have you see the documentation already on surface tracking: https://docs.pupil-labs.com/#surface-tracking ? You can add surfaces based on fiducial markers present in the world video. You can edit existing surfaces within the world video directly. 

Please could you clarify your question? It is not entirely clear what you are trying to achieve.

**Comment by portas525** [comment]
Hello,
the problem is, that when you have multiple surfaces you are not able or it is very hard to find or to highlight a specific surface. therefore it would be nice if someone can implement a button to easily choose the surface you like to edit. E.g. when you have 10 surfaces at one scene it is hard to find the right point to edit e.g. suface 5. Here it will be fine when you have a button to select each surface (1 = surface 1, 2 = surface 2, aso.). I hope that I wrote my question more understandable. 
Best 
portas525

**Comment by papr** [comment]
Starting in #1591, clicking the `edit surface` button will always enable the `edit` mode for the surface visible on top. While the `edit` mode is active, the surface will show handles to move the corner vertices. It is also supported to enable the `edit` mode for multiple surfaces together, such that one can create surfaces that precisely share an edge.
